### PR TITLE
Key value pair format change

### PR DESCRIPTION
**Pull request by gau94rav**
**Problem**
Tags filter uses `value_key` format for URL which can easily break with a sql layer returns `_` inside it's key and value names (value_1_key_1).

**Solution**
Replace the hacky pair system to use stringified objects in URL for easy parsing.

### CODE REVIEW

File: visualizations/frontend/filters/v2/TTagsInput.vue [hunk 1]
```diff
@@ -1,145 +1,144 @@
...
+              this.tempTags = [...tags, ...this.tempTags]; // Without defaults base-multiple-value-input cannot show selected tags.
+              this.selectedTags = [...value, ...this.selectedTags]; // base-multiple-value-input needs exact value instead of key:val pairs.
+            } else {
+              this.selectedTags = [...tags, ...this.selectedTags];
+            }
+          }
+        } catch (error) {
+          console.log(error);
+          console.log("Failed to parse tags data");
+        }
```
Comment (gau94rav) [inline visualizations/frontend/filters/v2/TTagsInput.vue:115]: Change.

File: visualizations/frontend/filters/v2/TTagsInput.vue [hunk 2]
```diff
@@ -1,145 +1,144 @@
...
+          if (!this.firstKey) {
+            break;
+          }
+          values[this.firstKey] = [item, ...(values[this.firstKey] || [])];
+        } else {
+          values[item.key] = [item.value, ...(values[item.key] || [])];
+        }
+      }
+      this.setFilterValue("selected_items", JSON.stringify(values));
+    },
```
Comment (gau94rav) [inline visualizations/frontend/filters/v2/TTagsInput.vue:131]: Change 2.

File: visualizations/frontend/filters/v2/TTagsInput.vue (hunk 1 above)
Comment (msorber) [reply visualizations/frontend/filters/v2/TTagsInput.vue:115]: Does this rise to the point of using a console.error() instead of console.log()?

**Review by msorber — APPROVED** [review]

File: visualizations/frontend/filters/v2/TTagsInput.vue (hunk 1 above)
Comment (gau94rav) [reply visualizations/frontend/filters/v2/TTagsInput.vue:115]: Yes, because it means component failed entirely. updating.

**Review by msorber — APPROVED** [review]
LGTM! This is an excellent approach, maybe you could suggest that we make it the default in the core at the next frontend meeting?

File: visualizations/frontend/filters/v2/TTagsInput.vue [hunk 3]
```diff
@@ -1,145 +1,144 @@
...
+            let tags = value.map((v) => {
+              return {
+                key,
+                value: v,
+              };
+            });
+            if (this.isSingleTag) {
+              // base-multiple-value-input is used for single tag filter.
+              this.tempTags = [...tags, ...this.tempTags]; // Without defaults base-multiple-value-input cannot show selected tags.
+              this.selectedTags = [...value, ...this.selectedTags]; // base-multiple-value-input needs exact value instead of key:val pairs.
```
Comment (sareg0) [inline visualizations/frontend/filters/v2/TTagsInput.vue:107]: Is `value` normally a string? If yes, would the end result be what you expect I would expect the string to be split like so: 

```
const aValue = "some_value" 
const myArray = [...aValue] 
console.log(myArray)
Array(10) [ "s", "o", "m", "e", "_", "v", "a", "l", "u", "e" ]
```

**Review by sareg0 — COMMENTED** [review]
@gau94rav would this PR need to be merged before the one in topcoat-core?

**Review by sareg0 — APPROVED** [review]

**Pull request merged**